File: 932-monotonic-array/monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        return inc(nums) || dec(nums);
    }
    int inc(vector<int>& nums){
        int prev = nums[0];
        for(int num : nums){
            if(num < prev){
                return false;
            }
            prev = num;
        }
        return true;
    }

    int dec(vector<int>& nums){
        int prev = nums[0];
        for(int num : nums){
            if(num > prev){
                return false;
            }
            prev = num;
        }
        return true;
    }
};

File: 932-monotonic-array/monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        def inc(nums):
            high = -float(inf)
            for num in nums:
                if num <high:
                    return False
                high = num
            return True
        
        def dec(nums):
            low = float(inf)
            for num in nums:
                if num >low:
                    return False
                low = num
            return True
        
        return inc(nums) or dec(nums)